addon/components/frost-popover.js

```diff
@@ -1,65 +1,96 @@
-
-  event: 'manual',
-  spacing: Ember.computed.alias('offset'),
-  typeClass: 'frost-popover',
-  visibility: false,
-
-  style: Ember.computed(function () {
-    return Ember.String.htmlSafe('display:none;')
-  }),
-
-  setup: Ember.on('init', function () {
-    const parentView = this.get('parentView')
-    parentView.on('click', (event) => {
-      if (!Ember.ViewUtils.isSimpleClick(event)) {
-        return true
-      }
-
-      this.toggleProperty('visibility')
-
-      if (this.get('visibility')) {
-        Ember.run.next(this, () => {
-          $('html').on('click.container.' + this.get('elementId'), (event) => {
-            let popover = Ember.$('.tooltip-frost-popover')
-            if (!popover.is(event.target) && _.isEmpty(popover.has(event.target))) {
-              this.set('visibility', false)
-              $('html').off('click.container.' + this.get('elementId'))
-            }
-          })
+  visible: false,
+  event: 'click',
+  closest: false,
+  offset: 10,
+  position: 'bottom',
+  index: 0,
+  excludePadding: false,
+  classNameBindings: ['position', 'visible:visible:invisible'],
+  classNames: ['tooltip-frost-popover'],
+  didRender () {
+    Ember.run.next(() => {
+      const context = this
+      if (this.get('closest')) {
+        this.$().closest(this.get('target')).on(this.get('event'), function () {
+          context.send('togglePopover')
+        })
+      } else {
+        $(this.get('target')).on(this.get('event'), function () {
+          context.send('togglePopover')
         })
       }
     })
-  }),
-
-  update: Ember.observer('visibility', function () {
-    if (this.get('visibility') === true) {
-      const parentView = this.get('parentView')
-      const tooltip = parentView.get('tooltip')
-      if (tooltip) {
-        tooltip.destroy()
-      }
-      parentView.renderTooltip(this)
-    }
-  }),
-
-  teardown: Ember.on('willDestroyElement', function () {
-    $('html').off('click.container.' + this.get('elementId'))
-  }),
-
+  },
   actions: {
-    close (action) {
-      this.set('visibility', false)
-      $('html').off('click.container.' + this.get('elementId'))
-      if (_.isFunction(action)) {
-        action()
+    close () {
+      if (this.get('isDestroyed')) { return }
+      this.set('visible', false)
+    },
+    togglePopover () {
+      this.toggleProperty('visible')
+      let position = this.get('position')
+      let targetRect
+      // eslint-disable-next-line max-len
+      let targetElement = this.get('closest') ? this.$().closest(this.get('target'))[this.get('index')] : this.get('parentView').$(this.get('target'))[this.get('index')]
+      targetRect = {
+        top: targetElement.offsetTop,
+        left: targetElement.offsetLeft,
+        bottom: targetElement.offsetTop + targetElement.offsetHeight,
+        right: targetElement.offsetLeft + targetElement.offsetWidth,
+        width: targetElement.offsetWidth,
+        height: targetElement.offsetHeight
+      }
```

Comment:
After some serious thought, offset might not be best choice since it would require that the popover be within the same parent container. Using `getBoundingClientRect` would is the safest bet but it would mean the popover needs to have `fixed` position. What do you think? It would also make my job of adjusting the position to be constrained inside the viewport a little easier.